### PR TITLE
Add Spell Checker

### PR DESCRIPTION
**Pull request by bpaul4**
## Addresses Issue: 
#15 

## Changes proposed in this PR:
- Add CI check on spelling for Python files, documentation and general text files
- Adds common exceptions ("IDAES", "idaes", "Nd", "nd") and fixes a small batch of typos

Open question - how to perform spell check on Jupyter notebooks (.ipynb) while avoiding false positives from the "id" tag?

Ideas:
- skip .ipynb files entirely
- hard-code exceptions e.g. "ba"
- use `[default]` `extend-ignore-re` to skip all hexadecimal strings
- use `[default]` `extend-ignore-re` to skip individual lines in individual files

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.

2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: .github/workflows/typos.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+extend-exclude = [
+    "*.svg",
+    "*.png",
+    "*.json",
+    ]
+
+[default.extend-words]
+# Ignore IDAES
+IDAES = "IDAES"
+idaes = "idaes"
```
Comment (lbianchi-lbl) [inline .github/workflows/typos.toml]: I've just checked to be sure, and it seems corrections are not case-sensitive, so we don't need different capitalizations here.

File: .github/workflows/typos.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+[files]
+extend-exclude = [
+    "*.svg",
+    "*.png",
```
Comment (lbianchi-lbl) [inline .github/workflows/typos.toml]: I think PNG files, as binary files, should be skipped by default, so we should be able to remove this?

**Review by lbianchi-lbl — APPROVED** [review]
Looks good. I've pushed a few minor tweaks to the configuration and GHA settings.

**Review by ksbeattie — APPROVED** [review]
LGTM

**Pull request merged**